clang-tools-extra/test/clang-tidy/checkers/bsl-using-ident-unique-namespace.cpp

```diff
@@ -0,0 +1,81 @@
+// RUN: %check_clang_tidy %s bsl-using-ident-unique-namespace %t
+// #include <cstdint>
+
+using func = void (*)(int, int);	// Compliant
+
+namespace n1
+{
+	using func = void (*)(int, int);
+	void f1()
+	{
+		using func = void (*)(void); // Non-compliant, reuses func identifier declared
+			// in the same namespace
+	}
+
+	template <class T>
+	using ptr = T*;
+	// using ptr = int*;
+	using myptr = int*;
+
+	template <class T>
+	void f2() {
+		using ptr = T*;		// Non-compliant
+	}
+}
+
+namespace n2
+{
+	using func = void (*)(int, int); // Compliant, reuses func identifier but in another namespace
```

Comment:
Correct. Once thing that is missing here are these same tests but using a class/struct instead of a namespace.